waiter/src/waiter/middleware.clj

```diff
@@ -0,0 +1,31 @@
+;;
+;;       Copyright (c) 2018 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.middleware
+  (:require [clojure.core.async :as async]
+            [full.async :refer (<?? <? go-try)]
+            [waiter.async-utils :as au]
+            [waiter.utils :as utils]))
+
+(defn wrap-context
+  "Wraps a handler, ensuring the request and response (or exception) contains data in the context map."
```

Comment:
Why does `response` need to have this context, e.g. `descriptor`? Should something upstream rely on the presence of the descriptor?

Reply:
Not yet.